terraform/create.go

```diff
@@ -48,12 +61,108 @@ func (t *Terraform) Create() error {
 		"-var", fmt.Sprintf("db_instance_class=%s", t.config.DeploymentConfiguration.DBInstanceClass),
 		"-var", fmt.Sprintf("db_username=%s", t.config.DeploymentConfiguration.DBUserName),
 		"-var", fmt.Sprintf("db_password=%s", t.config.DeploymentConfiguration.DBPassword),
+		"-var", fmt.Sprintf("mattermost_download_url=%s", t.config.DeploymentConfiguration.MattermostDownloadURL),
+		"-var", fmt.Sprintf("mattermost_license_file=%s", t.config.DeploymentConfiguration.MattermostLicenseFile),
 		"-auto-approve",
 		"./terraform",
 	)
 	if err != nil {
 		return err
 	}
+
+	output, err := t.getOutput()
+	if err != nil {
+		return err
+	}
+
+	// Updating the config.json for each instance.
+	for _, ip := range output.InstanceIps.Value {
+		sshc, err := sshConn(ip)
+		if err != nil {
+			mlog.Error("error in getting ssh connection", mlog.String("ip", ip), mlog.Err(err))
+			continue
+		}
+		func() {
+			defer func() {
+				err := sshc.Close()
+				if err != nil {
+					mlog.Error("error closing ssh connection", mlog.Err(err))
+				}
+			}()
+
+			mlog.Info("Updating config", mlog.String("host", ip))
+			var dsn string
+			switch t.config.DeploymentConfiguration.DBInstanceEngine {
+			case "postgres":
+				dsn = "postgres://" + t.config.DeploymentConfiguration.DBUserName + ":" + t.config.DeploymentConfiguration.DBPassword + "@" + output.DBEndpoint.Value + "/" + t.config.DeploymentConfiguration.ClusterName + "db?sslmode=disable"
+			case "mysql":
+				dsn = t.config.DeploymentConfiguration.DBUserName + ":" + t.config.DeploymentConfiguration.DBPassword + "@tcp(" + output.DBEndpoint.Value + ")/" + t.config.DeploymentConfiguration.ClusterName + "db?charset=utf8mb4,utf8\u0026readTimeout=30s\u0026writeTimeout=30s"
+			}
+			mlog.Info("dsn: " + dsn) // TODO: remove this later.
+			for k, v := range map[string]interface{}{
+				".ServiceSettings.ListenAddress":       ":8065",
+				".ServiceSettings.LicenseFileLocation": "/home/ubuntu/mattermost.mattermost-license",
+				".ServiceSettings.SiteURL":             "http://" + ip + ":8065",
+				".SqlSettings.DriverName":              t.config.DeploymentConfiguration.DBInstanceEngine,
+				".SqlSettings.DataSource":              dsn,
+				".MetricsSettings.Enable":              true,
+				".PluginSettings.Enable":               true,
+				".PluginSettings.EnableUploads":        true,
+			} {
+				// TODO: wrap this functionality in a separate package.
+				session, err := sshc.NewSession()
+				if err != nil {
+					mlog.Error("failed to create session", mlog.Err(err))
+					return
+				}
+				func() {
+					defer func() {
+						err := session.Close()
+						if err != nil && err != io.EOF { // Somehow it gives an io.EOF error every time. TODO: need to debug.
+							mlog.Error("error closing ssh session", mlog.Err(err))
+						}
+					}()
+
+					var b bytes.Buffer
+					session.Stdout = &b
```

Comment:
I didn't understand why we are doing this. There is the same thing below L:156

Reply:
Ooh, that's a mistake. Removed. Thanks.